osbuild/__init__.py

```diff
@@ -72,7 +72,7 @@ def __enter__(self):
         return self.root
 
     def __exit__(self, exc_type, exc_value, exc_tb):
-        self.unmount()
+        pass
```

Comment:
`__del__` is unpredictable. I just put it in there as well to "make sure" we unmount if someone forgets.

I think it definitely needs to be in `__exit__` to keep `with` semantics correct. If you'd like to re-use the object, we should move mounting to `__enter__`.

Reply:
Hm, indeed, __del__ may not be called at all. Would it not then be more correct to move everything into __enter__()/__exit__(), or just as well make this a @contectmanager method rather than a class?

Reply:
A `@contextmanager` is the same thing with more convenience. I think the class reads better in this case and would move the initialization into `__enter__()`.

Reply:
Yup. But class is fair enough, I'll rework.